crawl-ref/source/skills.cc

```diff
@@ -1145,7 +1146,7 @@ pair<int,int> skill_level_to_diffs(skill_type skill, double amount, int scaled_t
         you_skill += ash_skill_point_boost(skill, you.skills[skill] * 10);
     }
     if (target == you_skill)
-        return make_pair(0, 0);
+        return skill_diff();
 
     // XXX: reducing with ash bonuses in play could lead to weird results.
     const bool reduced = target < you_skill;
```

Comment:
I've cleaned this up in the next commit, but one thing that may not have been entirely clear is that this function is also used when actually setting skill levels -- it's a refactored version of the old code that set skill levels by side effect, in order to do just the calculations with no side effects.  So when `reduced` (now `decrease_skill`) is true, it's computing how much xp/skp to remove to reach the target skill level.  The function I factored it out of is `set_skill_level`.